internal/ethapi/api.go

```diff
@@ -1701,13 +1729,18 @@ func (s *TransactionAPI) GetTransactionReceipt(ctx context.Context, hash common.
 
 	// Derive the sender.
 	signer := types.MakeSigner(s.b.ChainConfig(), header.Number, header.Time)
+	return marshalReceipt(receipt, blockHash, blockNumber, signer, tx, int(index)), nil
+}
+
+// marshalReceipt marshals a transaction receipt into a JSON object.
+func marshalReceipt(receipt *types.Receipt, blockHash common.Hash, blockNumber uint64, signer types.Signer, tx *types.Transaction, txIndex int) map[string]interface{} {
```

Comment:
I think we're using interfaces as response objects too much in this package. You don't need to worry about it in this PR but generally pointing out that it'd be nice to use proper types more.

Reply:
Please ignore this as per https://github.com/ethereum/go-ethereum/pull/27305#issuecomment-1660277875

Reply:
sgtm